atlassian_jwt_auth/key.py

```diff
@@ -72,6 +72,9 @@ class HTTPSPublicKeyRetriever(BasePublicKeyRetriever):
     """ This class retrieves public key from a https location based upon the
          given key id.
     """
+    # Use a static requests session, reused/shared by all instances of
+    # HTTPSPublicKeyRetriever:
+    _session = None
```

Comment:
I'd prefer to not make such a change in this pull request. This is arguably a breaking behaviour / a change.

Reply:
Can we create a `CacheControl` instance that we mount to sessions?

Reply:
We could, though I'm not sure it's desirable to be creating new sessions for each retriever instance either. The session holds on the keepalive connections to reuse them the next time, and possibly other things.

I think the primary thinking here is to have the default behavior share the entire, single session between all instances of the HTTPSPublicKeyRetriever.